src/ip_notifier.cpp

```diff
@@ -91,7 +91,7 @@ namespace libtorrent
 				{ cb(error_code(err, system_category())); });
 		}
 #else
-		TORRENT_UNUSED(cb);
+		cb(make_error_code(boost::system::errc::not_supported));
```

Comment:
I hope you don't mind another PR for that, CI is green already :)

btw, I think I have a solution for macOS